private/buf/cmd/buf/command/alpha/plugin/pluginpush/pluginpush.go

```diff
@@ -198,13 +184,31 @@ func run(
 	var nextRevision uint32
 	// TODO: Revisit if we decide to make revision part of plugin_version
 	currentRevision, err := service.GetLatestCuratedPlugin(ctx, pluginConfig.Name.Owner(), pluginConfig.Name.Plugin(), pluginConfig.PluginVersion)
+	var currentImageDigest string
 	if err != nil {
 		if connect.CodeOf(err) != connect.CodeNotFound {
 			return err
 		}
 		nextRevision = 0
 	} else {
 		nextRevision = currentRevision.Revision + 1
+		currentImageDigest = currentRevision.ContainerImageDigest
+	}
+	if currentImageDigest != plugin.ContainerImageDigest() {
```

Comment:
Small optimization but if CI/CD flows run buf alpha plugin push ..., this will prevent us pushing up unique tags for the same digest as the last run.